### DISCUSSION TITLE
That request type is not allowed: Invalid request type

### DISCUSSION THREAD

**Discussion by iRootPro** [harrisoncramer/gitlab.nvim#55] [Q&A] [answered]
Hi there!
I have a question ;-)
I am trying this plugin with a self-hosted gitlab instance, and when I run gitlab.toggle_disscussion (leader + gld) I get an error: This request type is not allowed: Invalid request type
how can i fix this?

**Comment by harrisoncramer** [comment, marked as answer]
Thank you for catching this. This error is what happens when the method (GET/POST/PUT) isn't correct.

My guess is that you have the latest Lua code, which sends a POST, but the Go binary hasn't been rebuilt, so the endpoint is still expecting a GET (this was changed recently). I need to adjust the build process so that the binary is force-rebuilt even when it already exists, when someone updates the plugin.

I've just pushed up those changes directly to the main branch. Can you update to the latest version of the plugin? It should automatically rebuild when you close and reopen Neovim, but if not, then run this:

```
:lua require("gitlab.server").build(true)
```

Does that address the problem?

**Comment by iRootPro** [comment]
I updated gitlab.nvim for the latest commit: **ac2118c**
but it does not work
I am trying to add a "true" argument to the build method:
`require("gitlab.server").build(true)`
This does not work for me.

**Reply by harrisoncramer** [reply]
What do you mean it does not work, are you encountering an error when rebuilding the binary? 

I'm fairly certain that an outdated server is the issue for you.

You may make sure that the old server is not running (make sure to kill it if it is, it runs on 21036 by default). Then in Neovim:

```
:lua require("gitlab.server").build(true)
:lua require("gitlab.server").start(function() print("Server started") end)
```

Once the server is started you can directly interact with it server like any other process outside of Neovim:

```
$ curl --header "PRIVATE-TOKEN: ${GITLAB_TOKEN}" localhost:21036/info
```

Does this return the correct information? If it does, you should be able to also hit the discussions endpoint with a POST:

```bash
$ curl -X POST --header "PRIVATE-TOKEN: ${GITLAB_TOKEN}" -d '{ "blacklist": [] }' localhost:21036/discussions
```

If this command works, then you should be able to open the discussions pane in Neovim without any issues. Does this command return the discussions info? You'll notice that if you hit this endpoint with a GET it returns the error you described:

```bash
$ curl -X GET --header "PRIVATE-TOKEN: ${GITLAB_TOKEN}" -d '{ "blacklist": [] }' localhost:21036/discussions # {"message":"That request type is not allowed","details":"Invalid request type","status":405}
```

**Reply by iRootPro** [reply]
I killed the PID listening on port 21036 and rerun Neovim and it works fine.
Thanks for the quick fix.
